### PR TITLE
Aggregate queries

### PR DESCRIPTION
**Pull request by bpandey-CS**
Calculates a simple aggregate query over a profile, as a KPI and writes as a custom metric

### CODE REVIEW

File: aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        String script = "filter(MAIL_STATUS.equalTo(\"MAIL CONTINUE\")).count()";
+//        profileData.filter(col("MAIL_STATUS").isin(new String[]{"MAIL CONTINUE"})).count()
+//        String script = "count()";
+        ProfileWindowedAggregation engine = new ProfileWindowedAggregation(profileData, "180 days");
+
+        AggregationResult result = engine.runWindowedAggregation(script);
+        Long firstRow = (Long) result.getDf().collectAsList().get(0).get(1);
+        System.out.println("========");
+        System.out.println(firstRow);
+        System.out.println("========");
```
Comment (mcriscolo-cs) [inline aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java:47]: ensure there is a dataset named "cohort" associated with this, you can find it being set on the engine in CampaignService.runAgg

Comment (bpandey-CS) [reply aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java:47]: agreed, given that this is being run on a single profile I did not see a reason to set it, also it seems from the ProfileScriptEngine that we can add as many datasets as we want and name them whatever(including "cohort"), and use them in the script. Should we keep it CampaignService agnostic or do we want to follow 1 cohort + 1 supplementary data format (as used in campaign services)

Comment (mcriscolo-cs) [reply aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java:47]: I think the initial example should follow the same logic as what we have in phoenix to ensure the same results against the same expressions. If endusers want to further customize that is on them.

Comment (bpandey-CS) [reply aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java]:
```
Usage: profiles-example kpi-query [-hV] -d=<windowDuration> [-ed=<endDate>]
                                  -p=<project> -ps=<profileSchemaName>
                                  -s=<script> [-sd=<startDate>]
                                  [-cf=<cohortFilters>]...
Calculating KPI using aggregate using from Profiles
   -cf, --cohortFilters=<cohortFilters>
                            Cohort Filter
  -d, --duration=<windowDuration>
                            Window Duration
  -ed, --endDate=<endDate>
                            End Date, Set the time-frame over which the KPI is
                              calculated
  -h, --help                Show this help message and exit.
  -p, --project=<project>   Cortex Project to use
  -ps, --profile=<profileSchemaName>
                            Profile Schema Name
  -s, --script=<script>     KPI Script
  -sd, --startDate=<startDate>
                            Start Date, Set the time-frame over which the KPI
                              is calculated
  -V, --version             Print version information and exit.
```

with the latest commit, scripts like below work:
```
kpi-query -p local -ps cvs -s "filter(MAIL_STATUS.equalTo(\"MAIL CONTINUE\")).count()" -d "180 days"
kpi-query -p local -ps cvs  -d "180 days" -cf "MAIL_STATUS.equalTo(\"MAIL CONTINUE\")" -s "var profileCount = count(); var cohortCount = cohort.count(); cohortCount.divide(profileCount)" 
kpi-query -p local -ps cvs  -d "180 days" -cf "MAIL_STATUS.equalTo(\"MAIL CONTINUE\")" -cf "PEER_DESC.equalTo(\"MAPD\")" -cf "ELIGIBILITY.equalTo(\"ELIGIBLE\")" -s "var profileCount = count(); var cohortCount = cohort.count(); cohortCount.divide(profileCount)"
```

**Review by mcriscolo-cs — APPROVED** [review]

**Pull request merged**